### ISSUE TITLE
Regex in declaration-property-value-disallowed-list doesn't work

### ISSUE THREAD

**Issue by AndrejGajdos** [stylelint/stylelint#5540]
> Clearly describe the bug

Regex `^(?:\$|inherit|initial).+` doesn't work in `declaration-property-value-disallowed-list`. I want to disallow values for `color` css rule:

```
rgb(0,0,0)
#ffffff
green
```

but allow values:

```
$green
inherit
initial
```

> Which rule, if any, is the bug related to?

`declaration-property-value-disallowed-list`

> What code is needed to reproduce the bug?

e.g.

```css
$white = #ffffff;

a {
  color: $white;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "plugins": ['stylelint-scss'],
  "syntax": 'scss',
  "rules": {
    "declaration-property-value-disallowed-list": [{color: ["/^(?!(?:\$|inherit|initial)).+/"]}]
  }
}
```

> Which version of stylelint are you using?

`13.13.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"CLI with `stylelint "**/*.css" --config myconfig.json`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to SCSS variables."

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```shell
Unexpected value "$color-input-text" for property "color" (declaration-property-value-disallowed-list)
```

**Comment by jeddy3** [comment]
@AndrejGajdos Thanks for the report and for using the template.

You'll need to escape your slash, like so:

```json
{
  "rules": {
    "declaration-property-value-disallowed-list": [
      {
        "color": ["/^(?!(?:\\$|inherit|initial)).+/"]
      }
    ]
  }
}
```

[Demo here](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKA6A7FLIBsIAneFAYgDEqBubXA40ogcwCMAKAJgFZuAaFD36DeASlo5MkhiRQASAO4ALAJYAXGBPoRCslViUwi6rXh2MU+9SqT4JAX2wg+IAM4BPLGqQAPBOFdXZ3AILAAzFWZ-YDpMECIAV3wYVwwENFjcNNgwfCQiJDUVUIBaAAciCDKjNXcSgDdbBJgSqBVXW0IFGCgS-Ha1NNIAbUzcDMlx8bSZIZRRkAB6AD12AH4AQnX4DF2MOQAffUNjNSOsa1tRUQA6AGpFtIBdMdxHSZQXyXf7EHsgA).

Incidentally, it is only the `color` property you want to apply this limit to? Otherwise, using a combination of the  [color-named](https://stylelint.io/user-guide/rules/list/color-named/), [function-disallowed-list](https://stylelint.io/user-guide/rules/list/function-disallowed-list/) and [color-no-hex](https://stylelint.io/user-guide/rules/list/color-no-hex/) rules may be a better choice.

**Comment by AndrejGajdos** [comment]
@jeddy3 thanks for prompt answer :)
